### ISSUE TITLE
`no-invalid-triple-slash-reference` rule incorrectly warns on `lib` triple slash references in `.js` files

### ISSUE THREAD

**Issue by jaydenseric** [denoland/deno_lint#854]
TypeScript `lib` triple slash references should be allowed in `.js` files, but incorrectly result in `no-invalid-triple-slash-reference` rule warnings. See:

https://github.com/denoland/deno/discussions/11973

### Lint Name

`no-invalid-triple-slash-reference`

### Code Snippet

```js
// @ts-check
/// <reference lib="dom" />
/// <reference lib="deno.ns" />
```

### Expected Result

No lint errors.

### Actual Result

Lint errors:

```
This triple-slash reference directive is not allowed in JavaScript
In JavaScript only the `types` directive is allowed, like `/// <reference types="..." />`
```

### Additional Info

Until this is fixed, users can ignore the lint warnings:

```js
// @ts-check
// deno-lint-ignore no-invalid-triple-slash-reference
/// <reference lib="dom" />
// deno-lint-ignore no-invalid-triple-slash-reference
/// <reference lib="deno.ns" />
```

### Version

deno 1.13.2 (release, x86_64-apple-darwin)
v8 9.3.345.11
typescript 4.3.5

**Comment by kitsonk** [comment]
I don't know if we want to detect the ts-check or not, I suspect we would still get false positives if check-js is on in a config file. Maybe best to just "allow" the library reference in JS files.
